### PR TITLE
AuthUpgrade: Updated to use new Auth

### PR DESCRIPTION
**Pull request by BriWak**

### CODE REVIEW

File: test/controllers/GmpUsers.scala [hunk 1]
```diff
@@ -14,62 +14,65 @@
...
-    val sessionId = s"session-${UUID.randomUUID}"
-    lazy val request = FakeRequest().withSession(
-      SessionKeys.sessionId -> sessionId,
-      SessionKeys.token -> "RANDOMTOKEN",
-      SessionKeys.userId -> userId)
-    test(request)
-  }
-
-}
+///*
```
[the remark this answers is not in the document]
Comment (BriWak) [reply test/controllers/GmpUsers.scala]: yes, thought i already had. It's gone now

File: test/controllers/BulkRequestReceivedControllerSpec.scala [hunk 2]
```diff
@@ -70,118 +71,84 @@ class BulkRequestReceivedControllerSpec extends PlaySpec with OneServerPerSuite
...
         "respond with ok" in {
 
           when(mockSessionService.fetchGmpBulkSession()(Matchers.any(), Matchers.any())).thenReturn(Future.successful(Some(gmpBulkSession)))
           when(mockBulkRequestCreationService.createBulkRequest(Matchers.any(),Matchers.any(),Matchers.any(),Matchers.any())).thenReturn(Left(bulkRequest1))
-          when(mockGmpBulkConnector.sendBulkRequest(Matchers.any())(Matchers.any(),Matchers.any())).thenReturn(Future.successful(OK))
-          withAuthorisedUser { user =>
-            getBulkRequestReceived(user) { result =>
+          when(mockGmpBulkConnector.sendBulkRequest(Matchers.any(),Matchers.any())(Matchers.any())).thenReturn(Future.successful(OK))
+
+          val result = TestBulkRequestReceivedController.get.apply(FakeRequest())
```
Comment (daniel-manning) [inline test/controllers/BulkRequestReceivedControllerSpec.scala]: you should never need a literal apply TestBulkRequestReceivedController.get(FakeRequest()) is synonymous, expanded by the compiler

File: test/controllers/auth/AuthActionSpec.scala [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+          val controller = new Harness(authAction)
+
+          val result = controller.onPageLoad()(FakeRequest("", ""))
+          status(result) mustBe OK
+          whenReady(result) {
+            _.header.headers("link") mustBe "someID"
+          }
+        }
+      }
+
```
Comment (daniel-manning) [inline test/controllers/auth/AuthActionSpec.scala]: Need a test for both PSA and PP, where it uses the PSA link by preference

File: test/controllers/auth/FakeAuthAction.scala [hunk 4]
```diff
@@ -16,9 +16,14 @@
...
-object GmpGovernmentGateway extends GovernmentGateway {
-  override val loginURL = ExternalUrls.signIn
-  override val continueURL = ExternalUrls.continue
+import scala.concurrent.Future
+
+object FakeAuthAction extends AuthAction {
+
+  override def invokeBlock[A](request: Request[A], block: AuthenticatedRequest[A] => Future[Result]): Future[Result] = {
+
+    block(AuthenticatedRequest("gmp/B1234567", request))
```
Comment (daniel-manning) [inline test/controllers/auth/FakeAuthAction.scala]: Needs to be generated as it is an open service

File: test/controllers/auth/GmpRegimeSpec.scala [hunk 5]
```diff
@@ -14,119 +14,134 @@
...
+//    "not logged-in users" must {
+//
+//      "redirect to the gg sign-in page" in {
+//        val result = TestController.testRoute.apply(FakeRequest())
+//        status(result) must be(SEE_OTHER)
+//        redirectLocation(result).get must include("/gg/sign-in")
+//      }
+//    }
+//  }
+//}
```
Comment (daniel-manning) [inline test/controllers/auth/GmpRegimeSpec.scala]: delete it

File: test/controllers/auth/GmpGovernmentGatewaySpec.scala [hunk 6]
```diff
@@ -1,32 +0,0 @@
...
-package controllers.auth
-
-import org.scalatestplus.play.{OneServerPerSuite, PlaySpec}
-
-class GmpGovernmentGatewaySpec extends PlaySpec with OneServerPerSuite {
-  
-  "GmpGovernmentGateway" must {
-    
-    "have values overridden" in {
-      GmpGovernmentGateway.loginURL must be(ExternalUrls.signIn)
```
Comment (daniel-manning) [inline test/controllers/auth/GmpGovernmentGatewaySpec.scala]: We need a test for urls that it sends you to login

File: test/controllers/BulkRequestReceivedControllerSpec.scala (hunk 2 above)
Comment (BriWak) [reply test/controllers/BulkRequestReceivedControllerSpec.scala]: Done now

File: test/controllers/auth/AuthActionSpec.scala (hunk 3 above)
Comment (BriWak) [reply test/controllers/auth/AuthActionSpec.scala]: Done

File: test/controllers/auth/GmpRegimeSpec.scala (hunk 5 above)
Comment (BriWak) [reply test/controllers/auth/GmpRegimeSpec.scala]: Must have missed this one. Deleted Now.

File: app/config/GmpModule.scala [hunk 7]
```diff
@@ -19,11 +19,9 @@ package config
 import play.api.inject.{Binding, Module}
 import play.api.{Configuration, Environment}
 import uk.gov.hmrc.http.{HttpGet, HttpPost, HttpPut}
-import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
 
 class GmpModule extends Module{
   override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = Seq(
-    bind[AuthConnector].to(GmpFrontendAuthConnector),
```
[the remark this answers is not in the document]
Comment (BriWak) [reply app/config/GmpModule.scala]: Done now

File: test/controllers/auth/FakeAuthAction.scala (hunk 4 above)
Comment (BriWak) [reply test/controllers/auth/FakeAuthAction.scala]: This isn't a nino, it's a psa id, does it still need to be generated?

Comment (BriWak) [reply test/controllers/auth/FakeAuthAction.scala]: Actually it serves no purpose in the tests, it just needs to be there, so i have just changed it to "testID".

File: test/controllers/auth/GmpGovernmentGatewaySpec.scala (hunk 6 above)
Comment (BriWak) [reply test/controllers/auth/GmpGovernmentGatewaySpec.scala]: @Froom2 All other issues have been addressed so if you can take a look at this in the morning it's then ready to go i think.

File: test/controllers/auth/FakeAuthAction.scala (hunk 4 above)
Comment (daniel-manning) [reply test/controllers/auth/FakeAuthAction.scala]: Yeah that's fine. If it is any form of identifier it needs to be generated, just so people googling their identifiers are not confused

**Pull request merged**